### ISSUE TITLE
Variable casing error incorrectly reported if it happens to match a keyword

### ISSUE THREAD

**Issue by BrianGarland** [halcyon-tech/vscode-rpgle#234]
A Variable name casing error is reported if you use the POS() keyword on a DS subfield and happen to have a variable named Pos (short for position).  
![image](https://github.com/halcyon-tech/vscode-rpgle/assets/14948327/5e1e52a6-6a68-4177-b048-27b7fcadfa8c)

**Comment by worksofliam** [comment]
@BrianGarland Please paste a snippet I can copy to recreate :)

**Comment by BrianGarland** [comment]
@worksofliam Here you go:

`    DCL-DS MagicDS QUALIFIED;
        Char      CHAR(5000) POS(1);
        Integer   INT(10) POS(1);
        SmallInt  INT(5) POS(1);
        BigInt    INT(20) POS(1);
        Float     FLOAT(4) POS(1);
        Date      DATE POS(1);
        Time      TIME POS(1);
        TimeStamp TIMESTAMP POS(1);
        Real      FLOAT(4) POS(1);
        Double    FLOAT(8) POS(1);
        Binary    IND POS(1);
    END-DS;

    DCL-S Pos         INT(5);
`

**Comment by worksofliam** [comment]
@BrianGarland I have tried to recreate this and I cannot as of 0.20.1. Please try again, and if you're still getting it, share a fuller example and also your `rpglint.json` file. Thanks

**Comment by BrianGarland** [comment]
With the latest of all extensions installed, I can still reproduce it.   

`**FREE
DCL-DS MagicDS QUALIFIED;
    Char CHAR(5000) POS(1);
END-DS;

DCL-S Pos INT(5);

MagicDS.Char = 'x';

RETURN;
`

`{                                                                        
    "indent": 4,                                                         
    "NoGlobalsInProcedures": false,                                      
    "PrettyComments": false,                                             
    "NoOCCURS": false,                                                   
    "UppercaseConstants": true,                                          
    "UselessOperationCheck": true,                                       
    "CopybookDirective": "include",                                      
    "StringLiteralDupe": false,                                          
    "IncorrectVariableCase": true,                                       
    "RequireBlankSpecial": false,                                        
    "NoUnreferenced": true,                                              
    "UppercaseConstants":true,                                           
    "SpecificCasing": [{"operation":"*DECLARE","expected":"*upper"},     
                        {"operation":"*BIF","expected": "*upper"}]       
}                                                                        `

**Comment by SJLennon** [comment]
@worksofliam 
If you are looking at this, I have a similar **minor** issue with the "main" keyword on the ctl-opt statement. It gets confused if my proc is named Main.. (It's not hard to work around, so it's minor.)
![image](https://github.com/halcyon-tech/vscode-rpgle/assets/67484051/9690c50e-8a05-4ab0-9bd4-4a2ad0431f82)
If I change it to something other than main it doesn't complain.
![image](https://github.com/halcyon-tech/vscode-rpgle/assets/67484051/181590e4-5fb9-4468-85b3-dfd2abb091bf)

Here's a small bit of code that makes it happen:
```
**free
ctl-opt debug  option(*nodebugio: *srcstmt) dftactgrp(*no) actgrp(*caller)
main(Main);
dcl-proc Main;
    dsply %CHAR(CalcDiscount(10000));
    dsply %char(CalcDiscount(1000));
    x = %TIMESTAMP(y);
    y = %TimeStamp(x);
    return;
end-proc;
```

**Comment by worksofliam** [comment]
@BrianGarland @SJLennon 

So it looks like my original claim that I fixed this was false. I actually wrote an invalid test which I didn't see until today.

Now, however, your issue has been fixed. The fix will be released when Code for IBM i 2.1.0 has been release - hopefully later this week.

**Comment by worksofliam** [comment]
0.21.0 is being deployed and the fix for this is included. Thanks!
